File: benchmark/output/python-pygame/openai/chain_reaction.py
import pygame
import sys
import random
import math

def rand_float(min_val, max_val):
    return random.uniform(min_val, max_val)

pygame.init()
screen = pygame.display.set_mode((800, 600))
pygame.display.set_caption("Chain Reaction")
clock = pygame.time.Clock()

# Game state variables
num_circles = 30
cx = [rand_float(50.0, 750.0) for _ in range(num_circles)]
cy = [rand_float(50.0, 550.0) for _ in range(num_circles)]
cradius = [15.0 for _ in range(num_circles)]
cstate = [0.0 for _ in range(num_circles)]
ctimer = [0.0 for _ in range(num_circles)]

font = pygame.font.Font(None, 36)

def distance(x1, y1, x2, y2):
    return math.sqrt((x2 - x1) ** 2 + (y2 - y1) ** 2)

running = True
while running:
    dt = clock.tick(60) / 1000.0

    for event in pygame.event.get():
        if event.type == pygame.QUIT:
            running = False
        elif event.type == pygame.MOUSEBUTTONDOWN:
            mx, my = pygame.mouse.get_pos()
            for i in range(num_circles):
                if cstate[i] == 0.0 and distance(mx, my, cx[i], cy[i]) < 15.0:
                    cstate[i] = 1.0
                    break

    for i in range(num_circles):
        if cstate[i] == 1.0:
            cradius[i] += 80.0 * dt
            ctimer[i] += dt
            if ctimer[i] > 1.0:
                cstate[i] = 2.0
            for j in range(num_circles):
                if cstate[j] == 0.0 and distance(cx[i], cy[i], cx[j], cy[j]) < cradius[i] + 15.0:
                    cstate[j] = 1.0

    screen.fill((0, 0, 0))
    exploded_count = 0
    for i in range(num_circles):
        if cstate[i] == 0.0:
            pygame.draw.circle(screen, (255, 255, 255), (int(cx[i]), int(cy[i])), 15)
        elif cstate[i] == 1.0:
            pygame.draw.circle(screen, (255, 255, 255), (int(cx[i]), int(cy[i])), int(cradius[i]))
        elif cstate[i] == 2.0:
            pygame.draw.circle(screen, (255, 255, 255), (int(cx[i]), int(cy[i])), 3)
            exploded_count += 1

    text = font.render(f"Exploded: {exploded_count}", True, (255, 255, 255))
    screen.blit(text, (10, 10))

    pygame.display.flip()

pygame.quit()
sys.exit()

File: vibe-lang/benchmark/output/lua-love/openai/chain_reaction.lua
local cx = {}
local cy = {}
local cstate = {}
local cradius = {}
local ctimer = {}

function love.load()
    love.window.setMode(800, 600)
    love.window.setTitle("Chain Reaction")

    for i = 1, 30 do
        cx[i] = love.math.random(50, 750)
        cy[i] = love.math.random(50, 550)
        cstate[i] = 0.0
        cradius[i] = 15.0
        ctimer[i] = 0.0
    end
end

function love.update(dt)
    for i = 1, #cx do
        if cstate[i] == 1.0 then
            cradius[i] = cradius[i] + 80.0 * dt
            ctimer[i] = ctimer[i] + dt
            if ctimer[i] > 1.0 then
                cstate[i] = 2.0
            end
            for j = 1, #cx do
                if cstate[j] == 0.0 then
                    local dx = cx[i] - cx[j]
                    local dy = cy[i] - cy[j]
                    local distance = math.sqrt(dx * dx + dy * dy)
                    if distance < cradius[i] + 15.0 then
                        cstate[j] = 1.0
                    end
                end
            end
        end
    end
end

function love.mousepressed(x, y, button)
    if button == 1 then
        for i = 1, #cx do
            local dx = cx[i] - x
            local dy = cy[i] - y
            local distance = math.sqrt(dx * dx + dy * dy)
            if distance < 15.0 then
                cstate[i] = 1.0
                break
            end
        end
    end
end

function love.draw()
    local explodedCount = 0
    for i = 1, #cx do
        if cstate[i] == 0.0 then
            love.graphics.circle("line", cx[i], cy[i], 15)
        elseif cstate[i] == 1.0 then
            love.graphics.circle("line", cx[i], cy[i], cradius[i])
        elseif cstate[i] == 2.0 then
            love.graphics.circle("fill", cx[i], cy[i], 3)
            explodedCount = explodedCount + 1
        end
    end
    love.graphics.print("Exploded Circles: " .. explodedCount, 10, 10)
end